identity/aziot-identityd/src/identity.rs

```diff
@@ -407,4 +414,393 @@ impl IdentityManager {
+                            HubDeviceInfo::new(&prev_device_info_path).map_err(Error::Internal)?;
+
+                        if prev_hub_device_info == Some(curr_hub_device_info) {
+                            let prev_settings = crate::configext::load_file(&prev_settings_path)
+                                .map_err(Error::Internal)?;
+                            let (_, prev_hub_modules, _) =
+                                crate::configext::prepare_authorized_principals(
+                                    &prev_settings.principal,
+                                );
+                            prev_hub_modules
+                        } else {
+                            std::collections::BTreeSet::default()
+                        }
+                    } else {
+                        std::collections::BTreeSet::default()
+                    };
+
+                if prev_module_set.is_empty() && current_module_set.is_empty() {
+                    return Ok(());
+                }
+
+                let hub_module_ids = self.get_module_identities().await?;
+
+                for m in hub_module_ids {
+                    if let aziot_identity_common::Identity::Aziot(m) = m {
+                        if let Some(m) = m.module_id {
+                            if !current_module_set.contains(&m) && prev_module_set.contains(&m) {
+                                self.delete_module_identity(&m.0).await?;
+                                log::info!("Hub identity {:?} removed", &m.0);
+                            } else if current_module_set.contains(&m) {
+                                if prev_module_set.contains(&m) {
+                                    current_module_set.remove(&m);
+                                    log::info!("Hub identity {:?} already exists", &m.0);
+                                } else {
+                                    self.delete_module_identity(&m.0).await?;
+                                    log::info!("Hub identity {:?} will be recreated", &m.0);
+                                }
+                            }
+                        } else {
+                            log::warn!("invalid identity type returned by get_module_identities");
+                        }
+                    }
+                }
+
+                for m in current_module_set {
+                    self.create_module_identity(&m.0).await?;
+                    log::info!("Hub identity {:?} added", &m.0);
+                }
+
+                let () = std::fs::write(prev_device_info_path, device_status)
+                    .map_err(|err| Error::Internal(InternalError::SaveDeviceInfo(err)))?;
+
+                let () = std::fs::write(prev_settings_path, &settings_serialized)
+                    .map_err(|err| Error::Internal(InternalError::SaveSettings(err)))?;
+            }
+            None => log::warn!("reconcilation skipped since device is not provisioned"),
```

Comment:
I think this should be `log::info`, since there's nothing unusual about skipping Hub reconciliation for an unprovisioned device.

Reply:
Fixed